cloud-azure/src/main/resources/templates/arm-v2-freeipa.ftl

```diff
@@ -285,7 +285,12 @@
                                <#else>
                                "diskSizeGB": "${instance.rootVolumeSize}",
                                "managedDisk": {
-                                    "storageAccountType": "${instance.attachedDiskStorageType}"
+                                   <#if instance.managedDiskEncryptionWithCustomKeyEnabled>
+                                   "diskEncryptionSet": {
+                                        "id": "${instance.diskEncryptionSetId}"
+                                   },
```

Comment:
In its current state this will fail with 100% certainty. The `apiVersion` of `Microsoft.Compute/virtualMachines` has to be raised to `2019-07-01` at least, that is the earliest version supporting the DES ID parameter. With that API version jump comes an unrelated backward incompatible change, namely the way how MSIs can be attached to the VM. See https://github.com/hortonworks/cloudbreak/pull/10371/files#diff-7757b349409b97e1702d8c2f4c06878edeecc3a9c1f2a32d33d93ad2dd8ba376 for more details.